static/src/javascripts/projects/commercial/modules/dfp/fill-advert-slots.ts

```diff
@@ -0,0 +1,61 @@
+import config from '../../../../lib/config';
+import { getBreakpoint } from '../../../../lib/detect';
+import { commercialFeatures } from '../../../common/modules/commercial/commercial-features';
+import { removeDisabledSlots } from '../remove-slots';
+import { Advert } from './Advert';
+import { dfpEnv } from './dfp-env';
+import { displayAds } from './display-ads';
+import { displayLazyAds } from './display-lazy-ads';
+import { setupPrebidOnce } from './prepare-prebid';
+import { queueAdvert } from './queue-advert';
+
+// Pre-rendered ad slots that were rendered on the page by the server are collected here.
+// For dynamic ad slots that are created at js-runtime, see:
+//  article-aside-adverts
+//  article-body-adverts
+//  liveblog-adverts
+//  high-merch
+const fillAdvertSlots = (): Promise<void> => {
```

Comment:
Worth making it an `async` method? I think VSCode can do it automagically.

Then you can `await Promise.all(dependencies)` and simply do early returns.